pkg/ordermanager/ordermanager.go

```diff
@@ -209,21 +212,38 @@ func (om *impl) initiateNewOrder(ctx context.Context, domains []string, owner in
 		if err != nil {
 			return ticket{}, err
 		}
-		resp, err := om.Client.HTTP01ChallengeResponse(chal.Token)
+		chalResp, err := om.Client.HTTP01ChallengeResponse(chal.Token)
 		if err != nil {
 			return ticket{}, err
 		}
 		path := om.Client.HTTP01ChallengePath(chal.Token)
-		om.Challenger.RegisterChallenge(path, resp)
+		om.Challenger.RegisterChallenge(path, chalResp)
 
 		eg.Go(func() error {
 			defer om.Challenger.UnregisterChallenge(path)
 
-			// TODO(mattmoor): Wait until we have successfully probed the
-			// challenge ourselves before "Accepting" to get positive hand-off
-			// to the routing layer that things have been successfully plumbed.
-			// something something wait.Until()
-			time.Sleep(2 * time.Second)
+			err = wait.Poll(2*time.Second, 120*time.Second, func() (bool, error) {
+				r, err := http.Get("http://" + z.Identifier.Value + path)
```

Comment:
The ingress can be configured to redirect HTTP request to HTTPS. LetsEncrypt supports this case https://letsencrypt.org/docs/challenge-types/#http-01-challenge while here the check will fail.

I am not sure for the HTTP redirect case, the "wait" function will fail with time error or other error. If it is timeout error, it is fine to handle the HTTP redirect in a separate PR. Otherwise, we may want to handle it in this PR as it is a performance downgrade.

Reply:
yea, I am thinking that if the ingress of the cluster is configured to direct `http:// + z.Identifier.Value + path` to HTTPS, then in your case, you won't be able to get the right response even when the endpoint to serve the HTTP challenge requests is ready. So you may want to check the response of `http:// + z.Identifier.Value + path`. If it is redirected to another URL (e.g. HTTPS), then send the request to the new URL.